src/valkey-cli.c

```diff
@@ -5239,13 +5239,30 @@ static int clusterManagerMoveSlot(clusterManagerNode *source,
         /* Inform the source node. If the source node has just lost its last
          * slot and the target node has already informed the source node, the
          * source node has turned itself into a replica. This is not an error in
-         * this scenario so we ignore it. See issue #9223. */
+         * this scenario so we ignore it. See issue #9223.
+         *
+         * Another acceptable error can arise now that the primary pre-replicates 
+         * `cluster setslot` commands to replicas while blocking the client on the
+         * primary. This change enhances the reliability of `cluster setslot` in
+         * the face of primary failures. However, while our client is blocked on
+         * the primary awaiting replication, the primary might become a replica
+         * for the same reason as mentioned above, resulting in the client being
+         * unblocked with the role change error. */
         success = clusterManagerSetSlot(source, target, slot, "node", err);
-        const char *acceptable = "ERR Please use SETSLOT only with masters.";
-        if (!success && err && !strncmp(*err, acceptable, strlen(acceptable))) {
-            zfree(*err);
-            *err = NULL;
-        } else if (!success && err) {
+        if (!success && err) {
+            const char *acceptable[] = {
+                "ERR Please use SETSLOT only with masters.",
+                "UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)"};
```

Comment:
We tell end users not to take a dependency on the full string, just the code "UNBLOCKED". Should we do that here as well so we can change the rest of the language later?

Reply:
That is a good point. Let me fix this one first but we will likely need to scrub valkey-cli.c too (in future PRs)

Reply:
That sounds fine to me.

Reply:
It's trivial to change it to just "UNBLOCKED" I believe, so no need to postpone it to a different PR? The strncmp below just checks that err starts with one of these strings if I read it correctly.

```suggestion
                "UNBLOCKED"};
```